packages/sdk-v2/src/sdk/base/sdk.ts

```diff
@@ -87,14 +92,30 @@ export class BaseSDK {
   getContract(contractName: string): Contract;
   getContract(contractName: string) {
     if (!this.contracts[contractName]) return;
-
     switch (contractName) {
-      case "UBIScheme":
-        return new Contract(
+      case "UBIScheme": {
+        const contract = new Contract(
           this.contracts["UBIScheme"],
           CONTRACT_TO_ABI["UBIScheme"].abi,
           this.signer || this.provider
         ) as UBIScheme;
+
+        // for a divvi campaign, we add a suffix used registering a claimer for referal
+        const encodeFunctionDataPrev = contract.interface.encodeFunctionData.bind(contract.interface) as any;
+
+        contract.interface.encodeFunctionData = function (
```

Comment:
this should have been done only where we use the ubi contract and only when on celo.
you can keep it here, but see if you can add a condition to check the env is on celo

Reply:
That was where I initially did it, however, because the hooks used are called repeatedly due to useDapps refresh rate, the encodeFunctionData was called repeatedly resulting in 20+ suffixes added for the claim

Reply:
added check for current network